### ISSUE TITLE
Etiquetas para el modelo 0 (clasificador)

### ISSUE THREAD

**Issue by gitierrez** [mateoarv/kaggle-elo#7]
- [ ] Aplicación del algoritmo que permita reetiquetar las observaciones en dos grupos: "Normales" y "Anormales". Apenas tengas estas etiquetas porfa hace un push y me notificas para poder empezar a construir los modelos 1 (regresor de normales) y 2 (regresor de anormales), pues necesito estas etiquetas para separar el train.csv.

- [ ] Entrenamiento de un modelo que clasifique las etiquetas en función de las features del modelo baseline. Utiliza mi Pipeline (DateTransformer -> OneHotEncoder -> OrdinalEncoder -> DropTransformer -> Scaler) de preprocesamiento de los datos para esto. Podes utilizar cualquier tipo de modelo para esto. Intenta varios que tengan características geométricas diferentes (Regresión logística, KNN, CART, Random Forest, Adaboost, LGBM) o incluso podes proponer una arquitectura de red neuronal para el modelo (si podes utiliza [Keras ](https://keras.io/)para esto para que sea fácil de modificar la arquitectura). El objetivo es lograr un AUC de 0.95. Hace un buen [hyperparameter tuning](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RandomizedSearchCV.html) para que explotemos los modelos todo lo que podamos.

El issue se considera cerrado cuando hagas el push del .joblib con el modelo 0.

**Comment by gitierrez** [comment]
Para modelos desbalanceados podes utilizar cualquiera de los siguientes dos métodos:

Método 1: Over/UnderSampling
------------------------------------------------------------------------------------------------------------------
Balancear el dataset usando técnicas de Oversampling o Undersampling, para esto mira [este paquete](https://github.com/scikit-learn-contrib/imbalanced-learn), no lo he mirado pero se ve interesante.

La propuesta sería usar uno de los métodos de Over/Undersampling para balancear las clases y utilizar accuracy o AUC como métrica para optimizar el modelo.

Método 2: Cambiar la función a optimizar
------------------------------------------------------------------------------------------------------------------
Acordate que el "mejor" modelo es aquel que minimice una función de pérdida _L_ o maximice una función de puntaje _P_. La propuesta es que construyas 4 modelos, cada uno minimizando/maximizando una de las siguientes funciones:

* [Balanced accuracy score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.balanced_accuracy_score.html#sklearn.metrics.balanced_accuracy_score)
* [F-beta](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.fbeta_score.html#sklearn.metrics.fbeta_score)
* [Recall](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.recall_score.html#sklearn.metrics.recall_score)
* [AUC](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html#sklearn.metrics.roc_auc_score)
